ssl/record/rec_layer_d1.c

```diff
@@ -697,8 +702,10 @@ int do_dtls1_write(SSL_CONNECTION *sc, uint8_t type, const unsigned char *buf,
         /* if it went, fall through and send more stuff */
     }
 
-    if (len == 0)
-        return 0;
+    if (len == 0) {
+        *written = 0;
```

Comment:
Since TLS can call SSL_write with a buffer of zero bytes and it returns success I felt like for consistency SSL_write with DTLS should also return success when asked to write a buffer of zero size.